ktor-server/ktor-server-tests/jvm/test/io/ktor/tests/server/features/StaticContentTest.kt

```diff
@@ -154,6 +154,164 @@ class StaticContentTest {
+
+        application.routing {
+            static("firstgz") {
+                preCompressed(CompressedFileType.GZIP, CompressedFileType.BROTLI) {
+                    files(tempgz.parentFile)
+                }
+            }
+            static("firstbr") {
+                preCompressed(CompressedFileType.BROTLI, CompressedFileType.GZIP) {
+                    files(tempgz.parentFile)
+                }
+            }
+        }
+
+        handleRequest(HttpMethod.Get, "/firstgz/$publicFile") {
+            addHeader(HttpHeaders.AcceptEncoding, "gzip, br")
+        }.let { result ->
+            assertTrue(result.requestHandled)
+            assertEquals(cType, result.response.contentType())
+            assertEquals("gzip", result.response.headers[HttpHeaders.ContentEncoding].orEmpty())
+        }
+
+        handleRequest(HttpMethod.Get, "/firstbr/$publicFile") {
+            addHeader(HttpHeaders.AcceptEncoding, "gzip, br")
+        }.let { result ->
+            assertTrue(result.requestHandled)
+            assertEquals(cType, result.response.contentType())
+            assertEquals("br", result.response.headers[HttpHeaders.ContentEncoding].orEmpty())
+        }
+    }
+
+    @Test
+    fun testPreCompressedConfiguresImperatively() = withTestApplication {
+        val tempFile = File.createTempFile("testServeEncodedFile", ".dummy")
+        val publicFile = tempFile.nameWithoutExtension
+        val gzDir = File(tempFile.parentFile, "js").also { it.mkdirs() }
+        val brDir = File(tempFile.parentFile, "css").also { it.mkdirs() }
+
+        File(basedir, "features/StaticContentTest.kt".replaceSeparators()).run {
+            copyTo(File(gzDir, "$publicFile.js.gz"), true)
+            copyTo(File(brDir, "$publicFile.css.br"), true)
+        }
+
+        application.routing {
+            static("assets") {
+                preCompressed(CompressedFileType.GZIP) {
+                    files(gzDir)
+                }
+                preCompressed(CompressedFileType.BROTLI) {
+                    files(brDir)
+                }
+            }
+        }
+
+        handleRequest(HttpMethod.Get, "/assets/$publicFile.js") {
+            addHeader(HttpHeaders.AcceptEncoding, "gzip, br")
+        }.let { result ->
+            assertTrue(result.requestHandled)
+            assertEquals(ContentType.defaultForFileExtension("js"), result.response.contentType())
+            assertEquals("gzip", result.response.headers[HttpHeaders.ContentEncoding].orEmpty())
+        }
+
+        handleRequest(HttpMethod.Get, "/assets/$publicFile.css") {
+            addHeader(HttpHeaders.AcceptEncoding, "gzip, br")
+        }.let { result ->
+            assertTrue(result.requestHandled)
+            assertEquals(ContentType.defaultForFileExtension("css"), result.response.contentType())
+            assertEquals("br", result.response.headers[HttpHeaders.ContentEncoding].orEmpty())
+        }
+    }
+
```

Comment:
Could you please add a test with preCompressed nesting to ensure that it's working properly? This is the last change request, I believe :)

Reply:
Hi @cy6erGn0m , sorry for the delay, I actually wrote the test but never pushed it :(

Thanks for taking care of it and merging, it feels good to be able to contribute to Ktor!